Build failure: vmware-workstation

Steps to reproduce the behavior:

    Attempt to build vmware-workstation from the unstable branch using the following command:

sudo nixos-rebuild switch --flake ~/nixos#framework --show-trace

    Observe the build failure with the following error related to the --docdir option in the configure flags.

Build log

Running phase: updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
fixing libtool script ./ltmain.sh
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
configure flags: --disable-static --prefix=/nix/store/9lclanzcvqmx0jc15i06125phchhsbsp-gdbm-1.8.3 --bindir=/nix/store/9lclanzcvqmx0jc15i06125phchhsbsp-gdbm-1.8.3/bin --sbindir=/nix/store/9lclanzcvqmx0jc15i06125phchhsbsp-gdbm-1.8.3/sbin --includedir=/nix/store/f8yb3rhalwdpaig8lwxd4plgbqn8vagk-gdbm-1.8.3-dev/include --oldincludedir=/nix/store/f8yb3rhalwdpaig8lwxd4plgbqn8vagk-gdbm-1.8.3-dev/include --mandir=/nix/store/64jmdcjmdl94l4hbadwqsz7j8qg16dwb-gdbm-1.8.3-man/share/man --infodir=/nix/store/vs7bglvjl3pxp2ffc98v5w7xlinn0q5j-gdbm-1.8.3-info/share/info --docdir=/nix/store/9lclanzcvqmx0jc15i06125phchhsbsp-gdbm-1.8.3/share/doc/gdbm --libdir=/nix/store/6gzwn3dm5qzbvj5009ind2ihvi84csi2-gdbm-1.8.3-lib/lib --libexecdir=/nix/store/6gzwn3dm5qzbvj5009ind2ihvi84csi2-gdbm-1.8.3-lib/libexec --localedir=/nix/store/6gzwn3dm5qzbvj5009ind2ihvi84csi2-gdbm-1.8.3-lib/share/locale --enable-libgdbm-compat
configure: error: unrecognized option: --docdir=/nix/store/9lclanzcvqmx0jc15i06125phchhsbsp-gdbm-1.8.3/share/doc/gdbm
Try `./configure --help' for more information.

Additional context

I tried building the package on NixOS 24.05 with the following command, and the build succeeded without any issues:

`nix-shell -I nixpkgs=channel:nixos-24.05 -p vmware-workstation`

Notify maintainers

cc @cawilliamson @deinferno @vifino

Metadata

[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.45, NixOS, 24.11 (Vicuna), 24.11.20240814.c3aa7b8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/store/sfycwi72zfjsspidinx56ajaiffpyh17-source`

--- a/pkgs/applications/virtualization/vmware-workstation/default.nix
+++ b/pkgs/applications/virtualization/vmware-workstation/default.nix
@@ -79,20 +79,6 @@ let
     sha256 = "sha256-JSEW1gqQuLGRkathlwZU/TnG6dL/xWKW4//SfE+kO0A=";
   };
 
-  gdbm3 = gdbm.overrideAttrs (old: rec {
-    version = "1.8.3";
-
-    src = fetchurl {
-      url = "mirror://gnu/gdbm/gdbm-${version}.tar.gz";
-      sha256 = "sha256-zDQDOKLii0AFirnrU1SiHVP4ihWC6iG6C7GFw3ooHck=";
-    };
-
-    installPhase = ''
-      mkdir -p $out/lib
-      cp .libs/libgdbm*.so* $out/lib/
-    '';
-  });
-
   vmware-unpack-env = buildFHSEnv rec {
     name = "vmware-unpack-env";
     targetPkgs = pkgs: [ zlib ];
@@ -114,7 +100,7 @@ stdenv.mkDerivation rec {
     libxslt
     libxml2
     libuuid
-    gdbm3
+    gdbm
     readline
     xz
     cups

